test/mainnet-fork/Uniswapv4.t.sol

```diff
@@ -0,0 +1,1971 @@
+
+        assertEq(_getCurrentTick(), -50);
+
+        IncreasePositionResult memory increaseResult = _mintPosition(-30, -10, 1_000_000_000e6, type(uint160).max, type(uint160).max);
+
+        assertEq(increaseResult.amount0Spent, 1_000_950.445137e6); // Expected 0
+        assertEq(increaseResult.amount1Spent, 0); // Expected 998_950.644702e6
+        assertEq(increaseResult.amount1Spent + increaseResult.amount0Spent, 1_000_950.445137e6); // Expected 998_950.644702e6
+
+        /******************************************************************************************/
+        /*** Backrun                                                                            ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -50);
+
+        uint256 amountOut2 = _externalSwap(_user, address(usdt), uint128(amountOut1));
+
+        assertEq(amountOut2, 2_901_232.701533e6);
+        assertEq(usdc.balanceOf(_user), 2_901_232.701533e6); // Gained 1_232 USDC.
+        assertEq(usdt.balanceOf(_user), 0);
+
+        /******************************************************************************************/
+        /*** Remove Liquidity                                                                   ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -8);
+
+        DecreasePositionResult memory decreaseResult = _decreasePosition(increaseResult.tokenId, 1_000_000_000e6, 0, 0);
+
+        assertEq(decreaseResult.amount0Received, 0); // Expected 0
+        assertEq(decreaseResult.amount1Received, 998_960.634310e6); // Expected 998_950.644701e6
+        assertEq(decreaseResult.amount0Received + decreaseResult.amount1Received, 998_960.634310e6); // Expected 998_950.644701e6, and lost 1,989 USD.
+    }
+
+    function test_uniswapV4_attack_priceAboveToBelow_defended() public {
+        // Setup the pool and the controller.
+        vm.startPrank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -200, 200, 20);
+        rateLimits.setRateLimitData(_DEPOSIT_LIMIT_KEY,  2_000_000e18, 0);
+        rateLimits.setRateLimitData(_WITHDRAW_LIMIT_KEY, 2_000_000e18, 0);
+        vm.stopPrank();
+
+        /******************************************************************************************/
+        /*** Get max amounts                                                                    ***/
+        /******************************************************************************************/
+
+        assertEq(_getCurrentTick(), -7);
+
+        ( uint256 amount0Forecasted, uint256 amount1Forecasted ) = _getIncreasePositionMaxAmounts(-30, -10, 1_000_000_000e6, 0.99e18);
+
+        uint256 amount0Max = (amount0Forecasted * 1e18) / 0.99e18;
+        uint256 amount1Max = (amount1Forecasted * 1e18) / 0.99e18;
```

Comment:
These should just use straight up forcasted amounts with zero slippage, this is what the planner will likely do